### PR TITLE
Fixes for python coverage and pytest and added OWNERS

### PR DESCRIPTION
**Pull request by wumaxd**
# Changes

* Only install the requirements when the file is here and fix workspace -- following #459 
* Add OWNERS file
* Use latest tag for the python images as default

Now the default value for the python version is latest. This removes the burden to update to the newest python version.
I added myself also to the OWNERS file to drive the development of the tasks. Further, for the tests currently one of my repos is used and I wanted to be updated if there are issues. 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Yaml file complies with [yamllint](https://github.com/adrienverge/yamllint) rules.
- [x] Complies with [Catalog Orgainization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

Signed-off-by: wumaxd <wurzer.maxi@gmx.de>

### CODE REVIEW

**Comment by wumaxd** [comment]
> /ok-to-test
> You need to be part of the organization to be able to be in an OWNERs file. See [tektoncd/community#116](https://github.com/tektoncd/community/pull/116) as an example of PR to tektoncd/community to add yourself to the organization.

@vinamra28 Thanks! I have created the PR.
Would appreciate if you add yourself as well to the OWNERS file of these task, as you are also driving them forward 😄

**Comment by vinamra28** [comment]
> > /ok-to-test
> > You need to be part of the organization to be able to be in an OWNERs file. See [tektoncd/community#116](https://github.com/tektoncd/community/pull/116) as an example of PR to tektoncd/community to add yourself to the organization.
> 
> @vinamra28 Thanks! I have created the PR.
> Would appreciate if you add yourself as well to the OWNERS file of these task, as you are also driving them forward

Thanks @wumaxd, probably you can add my username :slightly_smiling_face:

File: task/pytest/0.1/pytest.yaml [hunk 1]
```diff
@@ -34,9 +34,16 @@ spec:
...
+        if [ -n "$(inputs.params.REQUIREMENTS_FILE)" ] && [ -e "$(inputs.params.REQUIREMENTS_FILE)" ];then
+          pip install -r $(inputs.params.SOURCE_PATH)/$(inputs.params.REQUIREMENTS_FILE)
+          pip show pytest || {
+            echo "###\nWarning: Pytest is missing in your requirements\n###";
+            pip install pytest
+          }
+        else
+          pip install pytest
+        fi
         pytest $(inputs.params.ARGS) $(inputs.params.SOURCE-PATH)
```
Comment (vinamra28) [inline task/pytest/0.1/pytest.yaml:49]: nit:
```suggestion
        pytest $(inputs.params.ARGS) $(inputs.params.SOURCE_PATH)
```

File: task/python-coverage/0.1/python-coverage.yaml [hunk 2]
```diff
@@ -37,8 +37,18 @@ spec:
...
+            pip install pytest
+          }
+          pip show coverage || {
+            echo "###\nWarning: Coverage is missing in your requirements\n###";
+            pip install coverage
+          }
+        else
+          pip install pytest
+          pip install coverage
         coverage run -m pytest $(inputs.params.ARGS) $(inputs.params.SOURCE-PATH)
```
Comment (vinamra28) [inline task/python-coverage/0.1/python-coverage.yaml:53]: nit:
```suggestion
        coverage run -m pytest $(inputs.params.ARGS) $(inputs.params.SOURCE_PATH)
```

File: task/python-coverage/0.1/python-coverage.yaml [hunk 3]
```diff
@@ -37,8 +37,18 @@ spec:
...
+            echo "###\nWarning: Pytest is missing in your requirements\n###";
+            pip install pytest
+          }
+          pip show coverage || {
+            echo "###\nWarning: Coverage is missing in your requirements\n###";
+            pip install coverage
+          }
+        else
+          pip install pytest
+          pip install coverage
```
Comment (vinamra28) [inline task/python-coverage/0.1/python-coverage.yaml:52]:
```suggestion
          pip install pytest coverage
```

File: task/pytest/0.1/pytest.yaml (hunk 1 above)
Comment (wumaxd) [reply task/pytest/0.1/pytest.yaml:49]: agree 👍

File: task/python-coverage/0.1/python-coverage.yaml (hunk 2 above)
Comment (wumaxd) [reply task/python-coverage/0.1/python-coverage.yaml:53]: agree 👍

**Comment by chmouel** [comment]
lgtm to me but let's hold for https://github.com/tektoncd/catalog/pull/459 so we don't have any conflict

/verify-owners
/hold

**Comment by vdemeester** [comment]
https://github.com/tektoncd/community/pull/168 and https://github.com/tektoncd/plumbing/pull/501 are required to fix peribolos :sweat_smile: (and thus adding @wumaxd to the org)

**Comment by chmouel** [comment]
/hold cancel
/lgtm


let's wait for those fixes and since #459 is merged we can go ahead

**Comment by vdemeester** [comment]
Ran it "manually" to quickly fix things, @wumaxd you should have received an invite to the org :wink:

**Comment by wumaxd** [comment]
> Ran it "manually" to quickly fix things, @wumaxd you should have received an invite to the org :wink: 

@vdemeester thanks! Got the invite 😊

**Pull request merged**